onnxruntime/core/providers/coreml/builders/impl/base_op_builder.cc

```diff
@@ -12,6 +13,15 @@ using namespace CoreML::Specification;
 namespace onnxruntime {
 namespace coreml {
 
+// Once all ops are supportted FP16, we can remove it. Before that, we keep a set of ops to
+// filter suppported ones.
+static std::set<std::string> Float16Ops = {
+    "Add", "Mul", "Sub", "Div", "Pow", "Sqrt", "Reciprocal",
+    "Sigmoid", "Tanh", "Relu", "LeakyRelu", "Concat", "GridSample", "GlobalAveragePool",
+    "Clip", "DepthToSpace", "Resize", "Slice", "Conv",
+    "ConvTranspose", "GlobalMaxPool", "Gemm", "MatMul",
+    "AveragePool", "MaxPool", "Reshape", "Split", "Transpose"};
```

Comment:
Is there test coverage of Slice, Split and Transpose? Even though they're not changing data there could be issues with the type handling to setup the CoreML layer that should have test coverage. 

e.g. Slice has this check https://github.com/microsoft/onnxruntime/blob/1bda91fc57d61367d245db808106687b43036bd1/onnxruntime/core/providers/coreml/builders/impl/slice_op_builder.cc#L224-L228